### PR TITLE
Add e2e test for operator

### PR DESCRIPTION
**Pull request by VaishnaviHire**
## How Has This Been Tested?
1. Testing steps:  https://github.com/opendatahub-io/opendatahub-operator/blob/3d53fece9b394fdb934145792795342ef3d3efb5/README.md#run-e2e-tests
2. Test Log:
```
make e2e-test
=== RUN   TestOdhOperator
=== RUN   TestOdhOperator/validate_operator_pod
=== RUN   TestOdhOperator/validate_operator_pod/Validate_ODH_Operator_pod
=== RUN   TestOdhOperator/validate_operator_pod/Validate_CRDs_owned_by_the_operator
=== RUN   TestOdhOperator/create
=== RUN   TestOdhOperator/create/e2e-test
=== RUN   TestOdhOperator/create/e2e-test/Creation_of_DataScienceCluster_instance
=== RUN   TestOdhOperator/create/e2e-test/Validate_deployed_applications
=== RUN   TestOdhOperator/create/e2e-test/Validate_Ownerrefrences_are_added
=== RUN   TestOdhOperator/create/e2e-test/Validate_Controller_Reverts_Updates
=== RUN   TestOdhOperator/create/e2e-test/Validate_Component_Enabled_field
--- PASS: TestOdhOperator (579.22s)
    --- PASS: TestOdhOperator/validate_operator_pod (74.33s)
        --- PASS: TestOdhOperator/validate_operator_pod/Validate_ODH_Operator_pod (10.11s)
        --- PASS: TestOdhOperator/validate_operator_pod/Validate_CRDs_owned_by_the_operator (60.89s)
    --- PASS: TestOdhOperator/create (504.89s)
        --- PASS: TestOdhOperator/create/e2e-test (502.00s)
            --- PASS: TestOdhOperator/create/e2e-test/Creation_of_DataScienceCluster_instance (10.30s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_deployed_applications (450.84s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Ownerrefrences_are_added (0.10s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Controller_Reverts_Updates (20.35s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Component_Enabled_field (20.41s)
PASS
ok      github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e     579.938s

```

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
PR to add CI job - https://github.com/openshift/release/pull/41508#pullrequestreview-1539984372

We can hold this, till the job gets added to retest

File: Makefile [hunk 1]
```diff
@@ -287,3 +291,7 @@ toolbox: ## Create a toolbox instance with the proper Golang and Operator SDK ve
 	$(IMAGE_BUILDER) stop opendatahub-toolbox ||:
 	toolbox rm opendatahub-toolbox ||:
 	toolbox create opendatahub-toolbox --image localhost/opendatahub-toolbox:latest
+
+.PHONY: e2e-test
+e2e-test: ## Run e2e tests for the controller
```
Comment (zdtsw) [inline Makefile:296]: should this one has dependency of "image deploy" ?

**Comment by zdtsw** [comment]
did local test:


`make image deploy e2e-test -e IMG=quay.io/wenzhou/opendatahub-operator:v0.0.73`

```
....build part....
go test ./tests/e2e/ -run ^TestOdhOperator -v --operator-namespace=opendatahub-operator-system "--skip-deletion=true" 
=== RUN   TestOdhOperator
=== RUN   TestOdhOperator/validate_operator_pod
I0721 10:56:30.386122  797008 request.go:682] Waited for 1.048342983s due to client-side throttling, not priority and fairness, request: GET:https://api.wenzhou-rosa-2.1roj.p3.openshiftapps.com:443/apis/authorization.openshift.io/v1?timeout=32s
=== RUN   TestOdhOperator/validate_operator_pod/Validate_ODH_Operator_pod
=== RUN   TestOdhOperator/validate_operator_pod/Validate_CRDs_owned_by_the_operator
=== RUN   TestOdhOperator/create
I0721 10:57:45.331958  797008 request.go:682] Waited for 1.047603999s due to client-side throttling, not priority and fairness, request: GET:https://api.wenzhou-rosa-2.1roj.p3.openshiftapps.com:443/apis/apiregistration.k8s.io/v1?timeout=32s
=== RUN   TestOdhOperator/create/e2e-test
=== RUN   TestOdhOperator/create/e2e-test/Creation_of_DataScienceCluster_instance
=== RUN   TestOdhOperator/create/e2e-test/Validate_deployed_applications
=== RUN   TestOdhOperator/create/e2e-test/Validate_Ownerrefrences_are_added
=== RUN   TestOdhOperator/create/e2e-test/Validate_Controller_Reverts_Updates
=== RUN   TestOdhOperator/create/e2e-test/Validate_Component_Enabled_field
--- PASS: TestOdhOperator (582.74s)
    --- PASS: TestOdhOperator/validate_operator_pod (75.49s)
        --- PASS: TestOdhOperator/validate_operator_pod/Validate_ODH_Operator_pod (10.13s)
        --- PASS: TestOdhOperator/validate_operator_pod/Validate_CRDs_owned_by_the_operator (61.27s)
    --- PASS: TestOdhOperator/create (507.26s)
        --- PASS: TestOdhOperator/create/e2e-test (503.65s)
            --- PASS: TestOdhOperator/create/e2e-test/Creation_of_DataScienceCluster_instance (10.43s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_deployed_applications (451.89s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Ownerrefrences_are_added (0.11s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Controller_Reverts_Updates (20.49s)
            --- PASS: TestOdhOperator/create/e2e-test/Validate_Component_Enabled_field (20.73s)
PASS
ok      github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e     582.758s
```

File: Makefile (hunk 1 above)
Comment (VaishnaviHire) [reply Makefile:296]: Adding that dependency will then remove the separation between deployment of operator and tests.

That means if someone deployed the operator through olm, they again will run `deploy` as part of `make e2e-test`.

**Review by zdtsw — APPROVED** [review]
/lgtm

**Pull request merged**